postgrest.cabal

```diff
@@ -24,6 +24,7 @@ Flag CI
 executable postgrest
   main-is:             PostgREST/Main.hs
   default-extensions:  OverloadedStrings, ScopedTypeVariables, QuasiQuotes, LambdaCase
+  ghc-options:         -threaded -rtsopts -with-rtsopts=-N
```

Comment:
I think `cabal check` will complain about having options such as `-O2` or `-Werror` in the cabal file. This would make hackage reject the package for upload.

Whereas cabal check is happy with these threading options, and I want to be sure even if someone builds with cabal that they will build with `-threaded`. So it seems like an OK split to have some options here and some in the stack file.